[Swift in WebKit] Convert TestPDFDocument from Obj-C++ to ObjSwift

rr-codes opened the pull request:
#### 2f65bcd3eb698bd2abd2335a47709469877aa70b
<pre>
[Swift in WebKit] Convert TestPDFDocument from Obj-C++ to ObjSwift
<a href="https://bugs.webkit.org/show_bug.cgi?id=294271">https://bugs.webkit.org/show_bug.cgi?id=294271</a>
<a href="https://rdar.apple.com/152980107">rdar://152980107</a>

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

To support using TestPDFDocument from Swift, it needs to stop using C++. To facilitate this,
the type no longer uses C++, and instead uses an Obj-C interface with a Swift implementation.
This has the added benefit of laying more foundation and infrastrcuture to make writing tests in Swift easier,
as well as setting good precedent.

* Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:

Compile out C++ specific parts of this file when it&apos;s being compiled for C/Swift.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPILibrary.xcconfig:

Add Swift support to TestWebKitAPILibrary.

* Tools/TestWebKitAPI/TestPDFDocument.h: Added.
* Tools/TestWebKitAPI/TestPDFDocument.swift: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DrawingToPDF.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFSnapshot.mm:
* Tools/TestWebKitAPI/cocoa/TestPDFDocument.h: Removed.
* Tools/TestWebKitAPI/cocoa/TestPDFDocument.mm: Removed.

Port the type to ObjSwift.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:

Canonical link: <a href="https://commits.webkit.org/296104@main">https://commits.webkit.org/296104@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/52da58b55584e6a0d22f57be5ae7558be995f711

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/107372 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/27054 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/17458 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/112586 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/57908 "Built successfully") 
| [✅ 🧪 bindings](https://ews-build.webkit.org/#/builders/9/builds/109336 "Passed tests") | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/27736 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/35554 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/81509 "Passed tests") | [⏳ 🧪 win-tests](https://ews-build.webkit.org/#/builders/Win-Tests-EWS "Waiting to run tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/110301 "Passed tests") | [✅ 🧪 ios-wk2](https://ews-build.webkit.org/#/builders/132/builds/21978 "Passed tests") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/96774 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/61884 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/21424 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/14905 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/57352 "Built successfully") | 
| [✅ 🛠 🧪 jsc](https://ews-build.webkit.org/#/builders/20/builds/99961 "Built successfully and passed tests") | [✅ 🧪 api-ios](https://ews-build.webkit.org/#/builders/13/builds/91347 "Passed tests") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/14937 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/115687 "Built successfully") | 
| [✅ 🛠 🧪 jsc-arm64](https://ews-build.webkit.org/#/builders/12/builds/105918 "Built successfully and passed tests") | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/34438 "Built successfully") | [✅ 🧪 mac-AS-debug-wk2](https://ews-build.webkit.org/#/builders/122/builds/25385 "Passed tests") | [✅ 🧪 gtk-wk2](https://ews-build.webkit.org/#/builders/1/builds/90548 "Passed tests") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/34814 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/93023 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/90285 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/23023 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/35201 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/12982 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/30182 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/34360 "Built successfully") | [❌ 🛠 mac-safer-cpp](https://ews-build.webkit.org/#/builders/120/builds/39901 "Failed to build and analyze WebKit") | [✅ 🛠 jsc-armv7](https://ews-build.webkit.org/#/builders/35/builds/130235 "Built successfully") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/34106 "Built successfully") | | [✅ 🧪 jsc-armv7-tests](https://ews-build.webkit.org/#/builders/25/builds/35425 "Passed tests") | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/37461 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/35767 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e614ea4b2d4c0e768af79056ad2a4bd67f06339d. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/1242632206f088f85114c94545559fed54d1fb84. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/13c34b1dd5695e9fce93564c43c95c4256b87175. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/38e5f5f05d7a94ba60d7d8d9734506319a2d5c75. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/d2615e558f4308153fc86a154427dd97c49958c4. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

pxlcoder reviewed (approved):
r=mews

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/ba0d53e865e7f8d1b0e7cfad7e13dc7cb9f61a56. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/b895379704f433a34e6b1d062c7eb634a8e16ada. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/2b90b03930dde505570eaffadf691c6da14dec36. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

webkit-early-warning-system commented:
Starting EWS tests for https://github.com/WebKit/WebKit/commit/e1067b954701f0e54748880d2647f1ee8e207123. Live statuses available at the PR page, https://github.com/WebKit/WebKit/pull/46553

pxlcoder commented on Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:
```diff
@@ -22,3 +22,6 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
+
+#import "../../TestPDFDocument.h"
```
Do we need to use a relative path?

rr-codes commented on Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
```diff
@@ -646,9 +647,9 @@
-		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
-		A176D24C2D6D2A4E00C63915 /* NetworkingExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2482D6D295400C63915 /* NetworkingExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
-		A176D24D2D6D2A4E00C63915 /* WebContentExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2492D6D295400C63915 /* WebContentExtension.appex */; platformFilter = ios; settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24B2D6D2A4E00C63915 /* GPUExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2472D6D295400C63915 /* GPUExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24C2D6D2A4E00C63915 /* NetworkingExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2482D6D295400C63915 /* NetworkingExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
+		A176D24D2D6D2A4E00C63915 /* WebContentExtension.appex in Embed Extensions */ = {isa = PBXBuildFile; fileRef = A176D2492D6D295400C63915 /* WebContentExtension.appex */; platformFilters = (ios, ); settings = {ATTRIBUTES = (CodeSignOnCopy, RemoveHeadersOnCopy, ); }; };
```
Xcode shenannigans 😓

rr-codes committed:
[Swift in WebKit] Convert TestPDFDocument from Obj-C++ to ObjSwift
https://bugs.webkit.org/show_bug.cgi?id=294271
rdar://152980107

Reviewed by Aditya Keerthi and Abrar Rahman Protyasha.

To support using TestPDFDocument from Swift, it needs to stop using C++. To facilitate this,
the type no longer uses C++, and instead uses an Obj-C interface with a Swift implementation.
This has the added benefit of laying more foundation and infrastrcuture to make writing tests in Swift easier,
as well as setting good precedent.

* Source/WebCore/PAL/pal/spi/cg/CoreGraphicsSPI.h:

Compile out C++ specific parts of this file when it's being compiled for C/Swift.

* Tools/TestWebKitAPI/Configurations/TestWebKitAPILibrary.xcconfig:

Add Swift support to TestWebKitAPILibrary.

* Tools/TestWebKitAPI/TestPDFDocument.h: Added.
* Tools/TestWebKitAPI/TestPDFDocument.swift: Added.
* Tools/TestWebKitAPI/Tests/WebKitCocoa/DrawingToPDF.mm:
* Tools/TestWebKitAPI/Tests/WebKitCocoa/PDFSnapshot.mm:
* Tools/TestWebKitAPI/cocoa/TestPDFDocument.h: Removed.
* Tools/TestWebKitAPI/cocoa/TestPDFDocument.mm: Removed.

Port the type to ObjSwift.

* Tools/TestWebKitAPI/TestWebKitAPI.xcodeproj/project.pbxproj:
* Tools/TestWebKitAPI/Tests/WebKit Swift/TestWebKitAPIBundle-Bridging-Header.h:

Canonical link: https://commits.webkit.org/296104@main

webkit-commit-queue commented:
Committed 296104@main (2f65bcd3eb69): <https://commits.webkit.org/296104@main>

Reviewed commits have been landed. Closing PR #46553 and removing active labels.
